core/src/main/java/ai/timefold/solver/core/impl/move/streams/dataset/AbstractForEachDataStream.java

```diff
@@ -15,13 +15,16 @@
 abstract sealed class AbstractForEachDataStream<Solution_, A>
         extends AbstractUniDataStream<Solution_, A>
         implements TupleSource
-        permits ForEachIncludingPinnedDataStream, ForEachExcludingPinnedDataStream, ForEachFromSolutionDataStream {
+        permits ForEachIncludingPinnedDataStream, ForEachExcludingPinnedDataStream {
 
     protected final Class<A> forEachClass;
+    private final boolean shouldIncludeNull;
 
-    protected AbstractForEachDataStream(DataStreamFactory<Solution_> dataStreamFactory, Class<A> forEachClass) {
+    protected AbstractForEachDataStream(DataStreamFactory<Solution_> dataStreamFactory, Class<A> forEachClass,
```

Comment:
We are implementing the builder pattern in some parts of the code. I wonder if we should adopt it as the default approach when there are three or more properties.